actor-model.mdx: re-written and expanded (#273)
statelyai/docs: docs/actor-model.mdx

---
title: The Actor model
---

The [Actor model](https://en.wikipedia.org/wiki/Actor_model) in computer science is a mathematical model of concurrent computation in which an "actor" is the basic building block.

> The actor model allows developers to build reliable message-based systems by using actors to communicate. State machines and statecharts can be modelled as actors, and can communicate with each other, and with other actors, in the same way.

Reviewer: ```suggestion
The actor model allows developers to build reliable message-based systems by using actors to communicate. State machines and statecharts can model the logic of actors. These actors can communicate with each other, and with other actors, in the same way.
```
